Source/DafnyCore/AST/TypeDeclarations/SubsetTypeDecl.cs

```diff
@@ -50,4 +50,48 @@ public enum WKind { CompiledZero, Compiled, Ghost, OptOut, Special }
   public virtual string GetDescription(DafnyOptions options) {
     return "subset type";
   }
+
+  public void AutoRevealDependencies(AutoRevealFunctionDependencies Rewriter, DafnyOptions Options, ErrorReporter Reporter) {
+    if (Witness is null) {
+      return;
+    }
+
+    var expressions = Constraint.SubExpressions.ToList().Concat(Witness.SubExpressions.ToList());
```

Comment:
It seems to me that there might be some shareable code between the function, method, constant, and subset type implementations of this method. We can leave that to a future PR, but I just wanted to note it for the record.

Reply:
It is just a sanity check to ensure that the feature is actually enabled. The program would successfully verify if functions were not made opaque, and the test would fail.

Reply:
Agreed.